include/ck/tensor_operation/gpu/device/device_base.hpp

```diff
@@ -5,17 +5,47 @@
 
 #include <string>
 #include <sstream>
+#include <regex>
+#include <optional>
 
 #include "ck/stream_config.hpp"
 
 namespace ck {
 namespace tensor_operation {
 namespace device {
 
+#define GET_OBJECT_NAME_IMLP                                                \
```

Comment:
Maybe we should put those macros into global scope?

Reply:
Extra headers are defined in `device_base.hpp`. Every class which needs to be inherited from `BaseOperator` requires to include `device_base.hpp`. This will be done explicitly or implicitly - i.e., through a chain of `#include`s. It seems to me that everything should be fine.

Reply:
These macros are specific to the `BaseOperator` class. I think they should live in `device_base.hpp`